Change all ARIA labels to translatable messages

wjhsf opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

Noticed when reviewing #1570 that the ARIA labels were hard-coded strings, rather than translatable messages. Did a quick search in the project to find all the other instances of that. Spotted a few small issues along the way, so I fixed those as well.

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Listed in commit history

# How to Test-Drive This PR

To test that the messages are now translatable:
- Update relevant translations in translation files
- Start dev server
- Observe that your changes are reflected on the page

To test other changes
- Start dev server
- Click "My account" button as guest. Should open login modal.
- Click "My account" button as registered user. Should go to account.
- Hitting space/enter on "My account" button should have the same result as clicking.
- Hitting space/enter on account menu icon should open the menu.
- Turn on VoiceOver
- View login screen.
- Tab over the 👁️‍🗨️ icon. Should say "show password".
- Click 👁️‍🗨️ icon. Should show password and now say "hide password".
- Tab focus on "My account" button. It should now say "My account, button"
  - It should probably be a link, rather than a button, but that's harder to change without breaking customers
- Tab focus on account dropdown icon. It should now say "Open account menu"

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

wjhsf committed:
Update field component aria label to translatable message.

Also a bit of cleanup!

wjhsf committed:
Convert account icon to more accessible button.

Also simplify guest/registered user logic.

wjhsf committed:
Convert my account menu aria-label to translatable message.

wjhsf committed:
Rename account menu vars for clarity.

wjhsf committed:
a11y: add space key handler

wjhsf committed:
Improve accessibility of popup account menu.

wjhsf committed:
Convert list menu aria label to translatable message.

wjhsf committed:
Convert pagination button aria labels to translatable messages.

wjhsf committed:
Convert carousel button aria labels to translatable messages.

wjhsf committed:
Update translations files.

wjhsf committed:
Update changelog.

wjhsf committed:
fix tests

alexvuong commented on packages/template-retail-react-app/app/components/_app/index.jsx:
```diff
@@ -257,18 +257,13 @@ const App = (props) => {
-            const path = buildUrl('/account')
-            history.push(path)
-        } else {
-            // if they already are at the login page, do not show login modal
```
This was a bug reported from UX awhile ago saying if a user is current in login page, the account click should not show the modal since user is already seeing the login form

wjhsf replied:
That bug is also handled by the `withRegistration` component. Although I noticed that here the logic handled `/login`, and the `withRegistration` component handles `/{locale}/login`, but what the app actually uses is `/global/{locale}/login`. I've changed the `withRegistration` component to be more flexible.

alexvuong commented on packages/template-retail-react-app/app/components/header/index.jsx:
```diff
@@ -154,51 +153,51 @@ const Header = ({
-                                    {...styles.arrowDown}
-                                    onMouseOver={onOpen}
-                                    tabIndex={0}
+                                    onClick={(e) => console.log('click', e.target)}
```
should we remove this console?

wjhsf committed:
Merge branch 'develop' into translatable-aria-label

Signed-off-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

wjhsf committed:
remove rogue console

wjhsf committed:
Use more helpful ARIA label.

wjhsf committed:
Fix login detection logic.

withRegistration component detects /{locale}/login, but app uses /global/{locale}/login and /login.

wjhsf committed:
Update changelog with `withRegistration` change.

alexvuong commented on packages/template-retail-react-app/app/components/with-registration/index.jsx:
```diff
@@ -13,20 +13,31 @@ import {useLocation} from 'react-router-dom'
             if (!customer.isRegistered) {
                 // Do not show auth modal if users is already on the login page
-                if (isLoginPage) {
+                if (isLoginPage(location, locale)) {
```
is locale being used?

wjhsf replied:
Not here, but it was in the original implementation, so I figured we'd keep it to make customizations easier.

alexvuong replied:
There is only one place that defines the isLoginPage which does not take locale. I don't think we need to do it here. 🤔 .

wjhsf replied:
Locale was used previously:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/9baf1c8cf344cdbe5c1a1c9ac6b7eb3976f68d70/packages/template-retail-react-app/app/components/with-registration/index.jsx#L22

Because our default implementation has multiple login pages ([/login](https://pwa-kit.mobify-storefront.com/login), [/en-GB/login](https://pwa-kit.mobify-storefront.com/en-GB/login), [/global/en-GB/login](https://pwa-kit.mobify-storefront.com/global/en-GB/login)), I decided to change the `isLoginPage` logic. But, to make it easier for customers to implement the previous logic (or any locale-dependent custom logic), I think it's reasonable to also provide `locale`.

wjhsf committed:
lint fixes

wjhsf committed:
Merge branch 'develop' into translatable-aria-label

Signed-off-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

wjhsf committed:
Merge branch 'develop' into translatable-aria-label

wjhsf committed:
Merge branch 'develop' into translatable-aria-label

Signed-off-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

wjhsf committed:
Merge branch 'develop' into translatable-aria-label
